pkg/prometheus/statefulset.go

```diff
@@ -339,10 +339,18 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapName
 		promArgs = append(promArgs,
 			fmt.Sprintf("-config.file=%s", path.Join(confOutDir, configEnvsubstFilename)),
 			fmt.Sprintf("-storage.tsdb.path=%s", storageDir),
-			"-storage.tsdb.retention="+p.Spec.Retention,
 			"-web.enable-lifecycle",
 			"-storage.tsdb.no-lockfile",
 		)
+		if version.Minor < 7 {
```

Comment:
I'd prefer to avoid if/else here. Let's do something along the lines of:

```go
retentionTimeFlag := "-storage.tsdb.retention="
if version.Minor >= 7 {
  retentionTimeFlag = "-storage.tsdb.retention.time="
}
promArgs = append(promArgs, retentionTimeFlag+p.Spec.Retention)
```

Reply:
Fixed